chips/veer_el2/src/chip.rs

```diff
@@ -1,90 +1,95 @@
 // Licensed under the Apache License, Version 2.0 or the MIT License.
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 // Copyright Tock Contributors 2022.
+// Copyright (c) 2024 Antmicro <www.antmicro.com>
 
 //! High-level setup and interrupt mapping for the chip.
 
+use crate::machine_timer::Clint;
 use core::fmt::Write;
 use core::ptr::addr_of;
 use kernel::platform::chip::{Chip, InterruptService};
-use kernel::utilities::cells::VolatileCell;
 use kernel::utilities::registers::interfaces::{ReadWriteable, Readable};
 use kernel::utilities::StaticRef;
 use rv32i::csr::{mcause, mie::mie, mip::mip, CSR};
+use rv32i::pmp::{simple::SimplePMP, PMPUserMPU};
 use rv32i::syscall::SysCall;
-use swerv::eh1_pic::{Pic, PicRegisters};
+
+use crate::pic::Pic;
+use crate::pic::PicRegisters;
 
 pub const PIC_BASE: StaticRef<PicRegisters> =
-    unsafe { StaticRef::new(0xF00C_0000 as *const PicRegisters) };
+    unsafe { StaticRef::new(0xf00c_0000 as *const PicRegisters) };
 
 pub static mut PIC: Pic = Pic::new(PIC_BASE);
 
-static mut TIMER0_IRQ: VolatileCell<bool> = VolatileCell::new(false);
-static mut TIMER1_IRQ: VolatileCell<bool> = VolatileCell::new(false);
-
-/// The UART interrupt line
-pub const IRQ_UART: u32 = 1;
-/// This is a fake value used to indicate a timer1 interrupt
-pub const IRQ_TIMER1: u32 = 0xFFFF_FFFF;
+pub const UART0_TX_WATERMARK: u32 = 1;
+pub const UART0_RX_WATERMARK: u32 = 2;
+pub const UART0_TX_EMPTY: u32 = 3;
+pub const UART0_RX_OVERFLOW: u32 = 4;
+pub const UART0_RX_FRAMEERR: u32 = 5;
+pub const UART0_RX_BREAKERR: u32 = 6;
+pub const UART0_RX_TIMEOUT: u32 = 7;
+pub const UART0_RX_PARITYERR: u32 = 8;
 
-pub struct SweRVolf<'a, I: InterruptService + 'a> {
+pub struct VeeR<'a, I: InterruptService + 'a> {
     userspace_kernel_boundary: SysCall,
     pic: &'a Pic,
-    scheduler_timer: swerv::eh1_timer::Timer<'static>,
-    mtimer: &'static crate::syscon::SysCon<'static>,
+    mtimer: &'static Clint<'static>,
     pic_interrupt_service: &'a I,
+    pmp: PMPUserMPU<4, SimplePMP<8>>,
 }
 
-pub struct SweRVolfDefaultPeripherals<'a> {
-    pub uart: crate::uart::Uart<'a>,
-    pub timer1: swerv::eh1_timer::Timer<'a>,
+pub struct VeeRDefaultPeripherals<'a> {
+    pub uart: crate::uart::UartType<'a>,
+    pub sim_uart: crate::uart::SimUartType<'a>,
 }
 
-impl<'a> SweRVolfDefaultPeripherals<'a> {
+impl<'a> VeeRDefaultPeripherals<'a> {
     pub fn new() -> Self {
         Self {
-            uart: crate::uart::Uart::new(crate::uart::UART_BASE),
-            timer1: swerv::eh1_timer::Timer::new(swerv::eh1_timer::TimerNumber::ONE),
+            uart: crate::uart::UartType::new(crate::uart::UART0_BASE, 125_000),
+            sim_uart: crate::uart::SimUartType::new(),
```

Comment:
Do we need both?

Reply:
I've realised we don't as `uart` was in fact unused as the testbench in VeeR only provides a special address which can be used as a virtual UART interface
Right now only `sim_uart` is used here.